target_chains/ethereum/contracts/forge-test/Pyth.WormholeMerkleAccumulator.t.sol

```diff
@@ -0,0 +1,424 @@
+
+        assertPriceFeedMessageStored(priceFeedMessages1[1]);
+        assertPriceFeedMessageStored(priceFeedMessages2[0]);
+        assertPriceFeedMessageStored(priceFeedMessages2[1]);
+    }
+
+    function testUpdatePriceFeedWithWormholeMerkleIgnoresOutOfOrderUpdateSingleCall()
+        public
+    {
+        PriceFeedMessage[]
+            memory priceFeedMessages1 = generateRandomPriceFeedMessage(1);
+        PriceFeedMessage[]
+            memory priceFeedMessages2 = generateRandomPriceFeedMessage(1);
+
+        // Make the price ids the same
+        priceFeedMessages2[0].priceId = priceFeedMessages1[0].priceId;
+        // Adjust the timestamps so the second timestamp is smaller than the first
+        // so it doesn't get stored.
+        priceFeedMessages1[0].publishTime = 10;
+        priceFeedMessages2[0].publishTime = 5;
+
+        bytes[] memory updateData = new bytes[](2);
+
+        uint8 depth = 0; // 1 messages
+        uint8 numSigners = 1;
+
+        updateData[0] = generateWhMerkleUpdate(
+            priceFeedMessages1,
+            depth,
+            numSigners
+        );
+        updateData[1] = generateWhMerkleUpdate(
+            priceFeedMessages2,
+            depth,
+            numSigners
+        );
+
+        uint updateFee = pyth.getUpdateFee(updateData);
+
+        pyth.updatePriceFeeds{value: updateFee}(updateData);
+
+        assertPriceFeedMessageStored(priceFeedMessages1[0]);
+    }
+
+    function testUpdatePriceFeedWithWormholeMerkleIgnoresOutOfOrderUpdateMultiCall()
+        public
+    {
+        PriceFeedMessage[]
+            memory priceFeedMessages1 = generateRandomPriceFeedMessage(1);
+        PriceFeedMessage[]
+            memory priceFeedMessages2 = generateRandomPriceFeedMessage(1);
+
+        // Make the price ids the same
+        priceFeedMessages2[0].priceId = priceFeedMessages1[0].priceId;
+        // Adjust the timestamps so the second timestamp is smaller than the first
+        // so it doesn't get stored.
+        priceFeedMessages1[0].publishTime = 10;
+        priceFeedMessages2[0].publishTime = 5;
+
+        (
+            bytes[] memory updateData,
+            uint updateFee
+        ) = createWormholeMerkleUpdateData(priceFeedMessages1);
+        pyth.updatePriceFeeds{value: updateFee}(updateData);
+        assertPriceFeedMessageStored(priceFeedMessages1[0]);
+
+        (updateData, updateFee) = createWormholeMerkleUpdateData(
+            priceFeedMessages2
+        );
+        pyth.updatePriceFeeds{value: updateFee}(updateData);
+        // Make sure that the old value is still stored
+        assertPriceFeedMessageStored(priceFeedMessages1[0]);
+    }
+
+    function isNotMatch(
+        bytes memory a,
+        bytes memory b
+    ) public pure returns (bool) {
+        return keccak256(a) != keccak256(b);
+    }
+
+    /// @notice This method creates a forged invalid wormhole update data.
+    function createAndForgeWormholeMerkleUpdateData(
```

Comment:
good test

Reply:
might be nice to document what the caller should pass as forgeItem (it's not super clear)

Reply:
Yeaah. thanks :bow: